addons/website/models/website.py

```diff
@@ -241,11 +245,138 @@ def unlink(self):
         attachments_to_unlink.unlink()
         return super(Website, self).unlink()
 
-    def create_and_redirect_to_theme(self):
+    def create_and_redirect_configurator(self):
         self._force()
-        action = self.env.ref('website.theme_install_kanban_action')
+        action = self.env.ref('website.start_configurator_act_url')
         return action.read()[0]
 
+    # ----------------------------------------------------------
+    # Configurator
+    # ----------------------------------------------------------
+    def _iap_website_service(self, route, params):
+        website_service_endpoint = self.env['ir.config_parameter'].sudo().get_param('website.website_service_endpoint', DEFAULT_ENDPOINT)
+        endpoint = website_service_endpoint + route
+        return iap_tools.iap_jsonrpc(endpoint, params=params)
+
+    @api.model
+    def configurator_init(self):
+        r = {}
+        company_id = self.get_current_website().company_id
+        r['features'] = self.env['website.configurator.feature'].search_read([], ['name', 'description', 'type', 'icon', 'website_types_preselection'])
+        r['logo'] = False
+        if company_id.logo and company_id.logo != company_id._get_logo():
+            r['logo'] = company_id.logo.decode('utf-8')
+        r['industries'] = self._iap_website_service('/website/industries', {'lang': self.env.user.lang})['industries']
+        return r
+
+    @api.model
+    def configurator_recommended_themes(self, industry_name, palette):
+        domain = [('name', '=like', 'theme%'), ('name', 'not in', ['theme_default', 'theme_common'])]
+        client_themes = [theme['name'] for theme in request.env['ir.module.module'].search_read(domain, ['name'])]
```

Comment:
In an other review (from an other PR) we told me that search_read + list comprehension was better/faster than search + mapped. But a search followed by a mapped like yours is indeed more readable

Reply:
that's the thing with being to told do something without knowing why.. some people call it a superstition ;)

`mapped()` is not evil, it's just a shortcut to dereferencing fields on records. However it can follow relationships on multiple levels, and at that point it quickly becomes more expensive that using a proper `search_read()` on the final model, where many optimisations are possible. But if you're just mapping a local field that is already prefetched in cache, it should not be a problem performance-wise - it's basically just a shortcut for `[x.field for x in records]`

Similarly, `records.filter()` is often frowned upon because it implements in-memory filtering of records, something that is often more effectively done by the database side at the source, where indexes can be used, etc. When working with tables that can have a significant volume, it is often better to execute a proper `search()` rather than `filter()` the data on the Odoo side.

To put it in a different way, you could `browse()`, `map()` and `filter()` you way through most of the database without ever using `search()`, and that would be disastrous in terms of performance. Yet there are cases where `map()` and `filter()` are exactly the right tool for the job.

As a rule of thumb, stay away from `map()` and `filter()` if that means traversing multiple relationships,  unless you've measured on real databases that it *is* faster than doing the alternative. 

For the other cases, use what feels most natural / elegant / idiomatic. If you're constructing dicts to represent records, you probably want to use `read()` or `search_read()` because that's exactly what they do. On the other hand if you only want a list of values for a single field, constructing those dicts and immediately unpacking them is more verbose, whereas `mapped()` does exactly what you want.

Reply:
Thank you for this very complete explanation! Indeed in this case mapped is more adapted than list comprehension

Reply:
> In an other review (from an other PR) we told me that search_read + list comprehension was better/faster than search + mapped.

That reasoning makes sense for the client, and why `search_read` was originally introduced: it's a single round-trip and atomic while search + read is two round-trips and two separate transactions.

On the Python side though, it doesn't make sense: `search_read` is literally implemented as a search, a read, and a bunch of post-processing (so the `read` bits are in the order specified by the `search`, also an advantage for the client as the ordering of `read`'s results is unspecified).